pkg/compiler/check.go

```diff
@@ -168,6 +168,17 @@ func (comp *compiler) checkStructFields(n *ast.Struct, typ, name string) {
 	if len(n.Fields) < 1 {
 		comp.error(n.Pos, "%v %v has no fields, need at least 1 field", typ, name)
 	}
+	for _, f := range n.Fields {
+		attrs := comp.parseAttrs(fieldAttrs, f, f.Attrs)
+		_, hasInDir := attrs[attrIn]
```

Comment:
We map bool attributes (without additional parameters) to 0 and 1, which is quite handy, we can do just:
```
if attrs[attrIn] + attrs[attrOut] + attrs[attrInOut] > 1 {...}
```